pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_cluster_settings failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_cluster_settings [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10657730?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10657730?buildTab=artifacts#/) on release-23.1 @ [dbf6ce8438742a73b9d17391c1ef868a4e7a9569](https://github.com/cockroachdb/cockroach/commits/dbf6ce8438742a73b9d17391c1ef868a4e7a9569):


```
=== RUN   TestTenantLogic_cluster_settings
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/47424cbdfa0dc4481b838b4a69ffa32f/logTestTenantLogic_cluster_settings2799836880
    test_log_scope.go:79: use -show-logs to present logs inline
[05:39:25] setting distsql_workmem='14829B';
[05:39:25] setting distsql_workmem='14829B';
    logic.go:2880: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/33/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:113: SELECT variable, value, default_value, origin FROM [SHOW ALL CLUSTER SETTINGS]
        WHERE variable IN ('sql.index_recommendation.drop_unused_duration')
        expected:
            sql.index_recommendation.drop_unused_duration  50s 168h0m0s external-override
            
        but found (query options: "") :
            sql.index_recommendation.drop_unused_duration  168h0m0s  168h0m0s  default
[05:39:25] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/33/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings with config 3node-tenant: 27 tests, 1 failures
    logic.go:3992: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/33/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:119: error while processing
    logic.go:3992: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/33/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:119: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/47424cbdfa0dc4481b838b4a69ffa32f/logTestTenantLogic_cluster_settings2799836880
--- FAIL: TestTenantLogic_cluster_settings (9.58s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_cluster_settings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29067

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_cluster_settings [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10672026?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10672026?buildTab=artifacts#/) on release-23.1 @ [dbf6ce8438742a73b9d17391c1ef868a4e7a9569](https://github.com/cockroachdb/cockroach/commits/dbf6ce8438742a73b9d17391c1ef868a4e7a9569):


```
=== RUN   TestTenantLogic_cluster_settings
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/47424cbdfa0dc4481b838b4a69ffa32f/logTestTenantLogic_cluster_settings2276529352
    test_log_scope.go:79: use -show-logs to present logs inline
[05:45:31] setting distsql_workmem='15063B';
[05:45:31] setting distsql_workmem='15063B';
    logic.go:2880: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3121/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:113: SELECT variable, value, default_value, origin FROM [SHOW ALL CLUSTER SETTINGS]
        WHERE variable IN ('sql.index_recommendation.drop_unused_duration')
        expected:
            sql.index_recommendation.drop_unused_duration  50s 168h0m0s external-override
            
        but found (query options: "") :
            sql.index_recommendation.drop_unused_duration  168h0m0s  168h0m0s  default
[05:45:32] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3121/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings with config 3node-tenant: 27 tests, 1 failures
    logic.go:3992: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3121/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:119: error while processing
    logic.go:3992: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3121/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:119: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/47424cbdfa0dc4481b838b4a69ffa32f/logTestTenantLogic_cluster_settings2276529352
--- FAIL: TestTenantLogic_cluster_settings (9.24s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_cluster_settings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_cluster_settings [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10681605?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10681605?buildTab=artifacts#/) on release-23.1 @ [82115b85a4c830bd6b32cfa9acca84a5d59d068e](https://github.com/cockroachdb/cockroach/commits/82115b85a4c830bd6b32cfa9acca84a5d59d068e):


```
=== RUN   TestTenantLogic_cluster_settings
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/dedbd8d3cba1e623e9cf0508efdf67be/logTestTenantLogic_cluster_settings2283125885
    test_log_scope.go:79: use -show-logs to present logs inline
    logic.go:2880: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9336/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:113: SELECT variable, value, default_value, origin FROM [SHOW ALL CLUSTER SETTINGS]
        WHERE variable IN ('sql.index_recommendation.drop_unused_duration')
        expected:
            sql.index_recommendation.drop_unused_duration  50s 168h0m0s external-override
            
        but found (query options: "") :
            sql.index_recommendation.drop_unused_duration  168h0m0s  168h0m0s  default
[17:30:53] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9336/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings with config 3node-tenant: 27 tests, 1 failures
    logic.go:3992: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9336/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:119: error while processing
    logic.go:3992: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9336/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:119: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/dedbd8d3cba1e623e9cf0508efdf67be/logTestTenantLogic_cluster_settings2283125885
--- FAIL: TestTenantLogic_cluster_settings (10.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_cluster_settings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/cluster_settings
+++ b/pkg/sql/logictest/testdata/logic_test/cluster_settings
@@ -110,7 +110,7 @@ ALTER TENANT ALL SET CLUSTER SETTING sql.index_recommendation.drop_unused_durati
 user root
 
 onlyif config 3node-tenant-default-configs
-query TTTT
+query TTTT retry
 SELECT variable, value, default_value, origin FROM [SHOW ALL CLUSTER SETTINGS]
 WHERE variable IN ('sql.index_recommendation.drop_unused_duration')
 ----

